tests/e2e/specs/blocks/recommendations/block-errors.spec.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * External dependencies.
+ */
+import {
+	createNewPost,
+	enablePageDialogAccept,
+	insertBlock,
+} from '@wordpress/e2e-test-utils';
+
+/**
+ * Internal dependencies.
+ */
+import {
+	activatePluginApiKey,
+	setSiteId,
+	startUpTest,
+} from '../../../utils';
+
+/**
+ * Tests for the Recommendations Block.
+ */
+describe( 'Recommendations Block', () => {
+	/**
+	 * Prevents browser from locking with dialogs, logs in to WordPress,
+	 * and activates the Parse.ly plugin.
+	 */
+	beforeAll( async () => {
+		enablePageDialogAccept();
+		await startUpTest();
+	} );
+
+	/**
+	 * Verifies that the block will display an error when an invalid Site ID is provided.
+	 */
+	it( 'Should display an error when an invalid Site ID is provided', async () => {
+		await activatePluginApiKey();
+		await createNewPost();
+		await insertBlock( 'Parse.ly' );
+
+		await page.waitForSelector( '.parsely-recommendations-error' );
+		const text = await page.$eval( '.parsely-recommendations-error', ( element ) => element.textContent );
+		expect( text ).toMatch( 'Access denied. Please verify that your Site ID is valid.' );
+	} );
+
+	/**
+	 * Verifies that the block will display an error when an empty Site ID is provided.
+	 */
+	it( 'Should display an error when an empty Site ID is provided', async () => {
+		await setSiteId( '' );
```

Comment:
Instead of introducing a new function, we could achieve this with current helpers: `deactivatePluginApiKey`

Reply:
In the case you mention, we could use `deactivatePluginApiKey()` but I'm unsure this is what we really want to do. Let me explain my reasoning:

The `setSiteId()` function was essentially introduced to:
- Be able to set a custom Site ID (which is not an option in `activatePluginApiKey()` where the ID is hardcoded).
- Start moving away from the `Plugin API Key` terminology towards `Site ID`, and this without doing a full change right now because I wanted to keep the PR focused.

Instead of using `deactivatePluginApiKey`, I would probably introduce a `setDefaultSiteId()` and `setEmptySiteId()` function to replace `activatePluginApiKey()` and `deactivatePluginApiKey()` respectively.

Alternatively, `setSiteId()` could have the default parameter `e2etest.example.com` if we want to have 2 functions (or even 1 function) instead of three. The bottom line is that for some tests, the Site ID will become important and we should be able to customize it.

This is an easy thing and I can do a PR for it if we want to.

Reply:
I believe that we should not introduce a duplicate function. This is internal code that we use for testing, so we don't have to keep strict backwards compatibility on the methods signature as we do in the rest of our code.

I like the proposal of using a default parameter on `setSiteId`, which would avoid having multiple functions. I think this is a tiny refactor, so we could very well do this in this PR if we want to move away from the API Key terminology.